micro-rdk/src/common/grpc/tests.rs

```diff
@@ -0,0 +1,240 @@
+    Ok((parts, body))
+}
+
+async fn build_request<M: Message + bytes::Buf + 'static>(
+    host: String,
+    path: String,
+    message: M,
+) -> Result<hyper::Request<BoxBody<M, Infallible>>, String> {
+    Ok(hyper::Request::builder()
+        .method(Method::POST)
+        .uri(host + &path)
+        .header(CONTENT_TYPE, "application/grpc")
+        .header(TE, "trailers")
+        .body(Full::new(message).boxed())
+        .unwrap())
+}
+
+#[cfg(feature = "camera")]
+async fn test_camera(
+    mut send_request: hyper::client::conn::http2::SendRequest<BoxBody<bytes::Bytes, Infallible>>,
+    host: &str,
+) -> Result<(), String> {
+    let get_image_path = "/viam.component.camera.v1.CameraService/GetImage";
+    // valid
+    let mut message = GetImageRequest::default();
+    message.name = "camera".to_string();
+    let message = encode_request(message).unwrap();
+
+    assert!(send_request.ready().await.is_ok());
+    let req = build_request(host.to_string(), get_image_path.to_string(), message)
+        .await
+        .map_err(|e| e.to_string())?;
+
+    let resp = send_request.send_request(req).await;
+    assert!(resp.is_ok());
+    let (parts, body) = check_response(resp.unwrap(), 0).await.unwrap();
+
+    let resp = GetImageResponse::decode(body.to_bytes().split_off(5));
+    assert!(resp.is_ok());
+    let resp = resp.unwrap();
+    assert_eq!(resp.mime_type, "image/jpeg");
+    assert_eq!(resp.image.len(), FAKE_JPEG.len());
+    assert_eq!(parts.status, 200);
+
+    // invalid
+    let mut message = GetImageRequest::default();
+    message.name = "non-existant-camera".to_string();
+    let message = encode_request(message).unwrap();
+
+    assert!(send_request.ready().await.is_ok());
+    let req = build_request(host.to_string(), get_image_path.to_string(), message)
+        .await
+        .map_err(|e| e.to_string())?;
+
+    let resp = send_request.send_request(req).await;
+    assert!(resp.is_ok());
+    let (_parts, _body) = check_response(resp.unwrap(), GrpcError::RpcUnavailable as i32)
+        .await
+        .unwrap();
+
+    // shouldn't work but does because the first proto field is `name` and we
+    // disregard `mime_type` anyway
+    let mut message = GetReadingsRequest::default();
+    message.name = "camera".to_string();
+    let message = encode_request(message).unwrap();
+
+    assert!(send_request.ready().await.is_ok());
+    let req = build_request(host.to_string(), get_image_path.to_string(), message)
+        .await
+        .map_err(|e| e.to_string())?;
+
+    let resp = send_request.send_request(req).await;
+    assert!(resp.is_ok());
+    let (parts, body) = check_response(resp.unwrap(), SUCCESS).await.unwrap();
+
+    let resp = GetImageResponse::decode(body.to_bytes().split_off(5));
+    assert!(resp.is_ok());
+    let resp = resp.unwrap();
+    assert_eq!(resp.mime_type, "image/jpeg");
+    assert_eq!(resp.image.len(), FAKE_JPEG.len());
+    assert_eq!(parts.status, 200);
+
+    Ok(())
```

Comment:
should I check the `mime_type` just to see if it's present? or enforce in the path handler that the mime_type is `image/jpeg`?

Reply:
following conversation with @stevebriskin, spinning this test snippet out as a separate bug ticket